app/services/targets/create_service.rb

```diff
@@ -11,13 +11,30 @@ def initialize(current_user, params)
     def call
       raise MaxAllowTargetsError unless targets.count < MAX_ALLOWED_TARGETS
 
-      targets.create!(params)
+      Target.transaction do
+        target = targets.create!(params)
+        find_match(target)
+        target
+      end
     end
 
     private
 
     def targets
       @targets ||= current_user.targets
     end
+
+    def find_match(target)
+      compatible_target = Target.within(target.radius, origin: [target.latitude, target.longitude])
+                                .where(topic_id: target.topic_id)
+                                .where.not(user_id: target.user_id)
+                                &.first
+
+      create_match(target, compatible_target) if compatible_target.present?
+    end
+
+    def create_match(target, compatible_target)
+      Match.create!(target: target, compatible_target: compatible_target)
```

Comment:
What happens if the user already has a match with the same target and/or person?

Reply:
@sebastiancaraballo check Juanma's answer to my question above

Reply:
In this case for the same Target will not happened, because Lean flag the matched Targets to `matched: true` and the TargetQuery will not return it, but for the same person, as I told to Agos, we will allow it for now and will create a new ticket to fix it later.. to avoid block this PR so long.